### PR TITLE
Add list by roles filters to appropriate entities. Update release notes.

### PR DESCRIPTION
**Pull request by plorenz**

### CODE REVIEW

File: doc/release-notes.md [hunk 1]
```diff
@@ -70,6 +71,67 @@ This adds operations to the `/services`, `/identities` and `/edge-routers` endpo
...
+         * roleSemantic. Optional. Defaults to AllOf if not specified. Indicates which semantic to use when evaluating role matches
+ 
+     * Endpoint: /services
+     * Query: GET /services now supports two new query paramters
+         * roleFilter. May be specified more than one. Should be an id or role specifier (ex: @38683097-2412-4335-b056-ae8747314dd3 or #sales)
+         * roleSemantic. Optional. Defaults to AllOf if not specified. Indicates which semantic to use when evaluating role matches
+         
+Note that a roleFilter should have one role specifier (like `@some-id` or `#sales`). If you wish to specify multiple, provide multiple role filters. 
+
+    /edge-routers?roleFilter=#sales@roleFilter=#us
```
Comment (andrewpmartinez) [inline doc/release-notes.md]: Shouldn't this be `?roleFilter=<val>&roleFilter=<val>` Looks like a @ instead of an &

File: doc/release-notes.md [hunk 2]
```diff
@@ -70,6 +71,67 @@ This adds operations to the `/services`, `/identities` and `/edge-routers` endpo
...
+ 
+     * Endpoint: /identities
+     * Query: GET /identities now supports two new query paramters
+         * roleFilter. May be specified more than one. Should be an id or role specifier (ex: @38683097-2412-4335-b056-ae8747314dd3 or #sales)
+         * roleSemantic. Optional. Defaults to AllOf if not specified. Indicates which semantic to use when evaluating role matches
+ 
+     * Endpoint: /services
+     * Query: GET /services now supports two new query paramters
+         * roleFilter. May be specified more than one. Should be an id or role specifier (ex: @38683097-2412-4335-b056-ae8747314dd3 or #sales)
+         * roleSemantic. Optional. Defaults to AllOf if not specified. Indicates which semantic to use when evaluating role matches
```
Comment (andrewpmartinez) [inline doc/release-notes.md]: question about 'allOf' and 'oneOf' are they case sensitive?

File: doc/release-notes.md (hunk 1 above)
Comment (plorenz) [reply doc/release-notes.md]: Fixed

File: doc/release-notes.md (hunk 2 above)
Comment (plorenz) [reply doc/release-notes.md]: I think they are. I'll see if I can tweak to be case insensitive.

Comment (plorenz) [reply doc/release-notes.md]: Changed to make them case insensitive

**Pull request merged**